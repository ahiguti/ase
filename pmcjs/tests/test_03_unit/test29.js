
use('Perl');
Perl.Use('FuncTest');
Perl.Functions['FuncTest'].foo('abc', 'def');
Perl.FunctionsLC['FuncTest'].foo('abc', 'def');
Perl.Methods['FuncTest'].foo('abc', 'def');
Perl.MethodsLC['FuncTest'].foo('abc', 'def');
Perl.Constants['FuncTest'].foo;
Perl.ConstantsLC['FuncTest'].foo;
var fooarr = Perl.Arrays['FuncTest'].fooarr;
for (var i = 0; i < fooarr.length; ++i) {
  print(fooarr[i]);
}
var foohash = Perl.Hashes['FuncTest'].foohash;
for (var i in foohash) {
  print(i + ": " + foohash[i]);
}
print(Perl.Scalars['FuncTest'].fooscalar);
Perl.Scalars['FuncTest'].fooscalar = 'def';
print(Perl.Scalars['FuncTest'].fooscalar);
