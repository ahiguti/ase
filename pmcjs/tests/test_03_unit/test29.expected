foo: argc=2
wantarray=
0: abc
1: def
foo: argc=2
wantarray=1
0: abc
1: def
foo: argc=3
wantarray=
0: FuncTest
1: abc
2: def
foo: argc=3
wantarray=1
0: FuncTest
1: abc
2: def
foo: argc=0
wantarray=
foo: argc=0
wantarray=1
10
20
30
a: 10
b: 20
abc
def
